icu4c/source/test/depstest/dependencies.txt

```diff
@@ -49,6 +50,20 @@ group: std_mutex
     std::condition_variable_any::condition_variable_any()
     std::condition_variable_any::~condition_variable_any()
 
+group: std_thread
+    std::thread::hardware_concurrency()
+    std::thread::join()
+    std::__throw_system_error(int)
+    std::thread::_State::~_State()
+    std::__throw_bad_alloc()
+    std::__throw_bad_function_call()    
+    pthread_create
+    "std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)())"
+    "std::__throw_length_error(char const*)"
+    "operator new(unsigned long)"
```

Comment:
We must not use global operator new, which also means no STL code that allocates memory.

Reply:
Done. Remove all thread related dependencies.